addons/point_of_sale/tests/test_point_of_sale_flow.py

```diff
@@ -1957,3 +1957,98 @@ def test_multi_exp_account_real_time(self):
 
         moves = self.env['account.move'].search([('ref', '=', f'pos_order_{order.id}')])
         self.assertEqual(len(moves), 2)
+
+    def test_multi_currency_journal(self):
+        """Verify that debit and credit are balanced for a company currency as EUR and a selling currency as USD."""
+
+        # Get currencies
+        currency_usd = self.env['res.currency'].search([('name', '=', 'USD')], limit=1)
+        currency_eur = self.env['res.currency'].search([('name', '=', 'EUR')], limit=1)
+        self.env.company.currency_id = currency_eur.id
+
+        # Set USD rate compared with EUR
+        currency_usd.rate_ids.create({
+            'company_rate': 2.0,
+            'currency_id': currency_usd.id,
+            'company_id': self.env.company.id
+        })
+
+        # Set USD to journals
+        self.env.company.bank_journal_ids.currency_id = currency_usd.id
+        journal_cash = self.env['account.journal'].search(
+            [('type', '=', 'cash'), ('company_id', '=', self.env.company.id)], limit=1
+        )
+        journal_cash.currency_id = currency_usd.id
+
+        journal_pos = self.env['account.journal'].create({
```

Comment:
Are the default journal an payment methods already using USD ?

Reply:
The default journal doesn't have a currency set. But we can just set its currency to USD, there is actually no need to create a new journal. (applied in next push)